### PR TITLE
fragment & lifecycle demo

### PR DESCRIPTION
**Pull request by ipcjs**
**Description**:
I wrote  `JavaFragment` by mimicking `JavaActivity`.
But it doesn't work correctly, AutoDispose doesn't `dispose` at the right time.

In a Fragment, how should I make AutoDispose work with `AndroidLifecycleScopeProvider`?

**Related issue(s)**:

### CODE REVIEW

**Comment by shaishavgandhi05** [comment]
Not sure I understand by "AutoDispose doesn't `dispose` at the right time. I tried out your sample and it seems to work fine. Once you go back from the activity, all subscriptions stop.

**Comment by hzsweers** [comment]
Agreed, would like you understand what you think isn't working correctly?

**Comment by shaishavgandhi05** [comment]
I think this is intended behavior. The lifecycle state in `onActivityCreated` is `CREATED`. If you check out the [LifecycleEventsObservable](https://github.com/uber/AutoDispose/blob/9313c3f30fc928f6b26beb79ac4b51cfd3cdb190/android/autodispose-android-archcomponents/src/main/java/com/uber/autodispose/android/lifecycle/LifecycleEventsObservable.java#L58) you'll see that AutoDispose maps between [Lifecycle states](https://developer.android.com/reference/androidx/lifecycle/Lifecycle.State) and the corresponding [Lifecycle.Event](https://developer.android.com/reference/androidx/lifecycle/Lifecycle.Event) so that it can reliably find the corresponding event to dispose off your subscription. 

The Lifecycle.State during `onCreate` is `INITIALIZED` which we map to `Lifecycle.Event.ON_CREATE` etc. In both `onStart` and `onActivityCreated` the Lifecycle.State is `CREATED` which would map it to an `Lifecycle.Event.ON_START`. That is my understanding and I will defer to @hzsweers if I got something wrong. 

The TL;DR of it is to use the overloaded `AndroidLifecycleScopeProvider.from(LifecycleOwner owner, Lifecycle.Event untilEvent)` to explicitly want a disposal in `onDestroy`. You can use it like so:
`AndroidLifecycleProvider.from(this, Lifecycle.Event.ON_DESTROY)`

**Comment by hzsweers** [comment]
The advised solution in SDK 28+ is to use the new view lifecycle of fragments. I'll try to put together something in the sample soon-ish

**Comment by hzsweers** [comment]
Closing this in favor of #260

**Comment by hzsweers** [comment]
In general - yes correct behavior would be to use viewLifecycleOwner
